core/templates/dev/head/components/share/SharingLinksDirective.js

```diff
@@ -22,39 +22,51 @@ oppia.directive('sharingLinks', [function() {
     scope: {
       layoutType: '@',
       layoutAlignType: '@',
+      shareType: '@',
       getTwitterText: '&twitterText',
-      getExplorationId: '&explorationId'
+      getExplorationId: '&explorationId',
+      getCollectionId: '&collectionId'
     },
     templateUrl: 'components/sharingLinks',
     controller: [
-      '$scope', '$window', 'oppiaHtmlEscaper', 'ExplorationEmbedButtonService',
+      '$scope', '$window', '$injector', 'oppiaHtmlEscaper',
```

Comment:
Ah, in that case, can we not use $injector and include the extra import where needed? We want the dependencies to be explicit in general, we use $injector only when this is not possible since it's a bit indirect.